dapps/marketplace/src/pages/identity/OAuthAttestation.js

```diff
@@ -92,11 +92,27 @@ class OAuthAttestation extends Component {
   }
 
   componentDidUpdate(prevProps, prevState) {
+    const { isMobile, provider } = this.props
+
     const didOpen = !prevProps.open && this.props.open,
       didChangeStage = prevState.stage !== this.state.stage
     if (this.inputRef && (didOpen || didChangeStage)) {
       this.inputRef.focus()
     }
+
+    if (provider === 'google' && !prevProps.isMobile && isMobile) {
+      // Check wether the current browser is WebView in order to block Google OAuth and warn the user
```

Comment:
When the profile page is being loaded in the mobile, It is always in a webview. User cannot access this page without a DApp browser. 

Instead of checking these, we could just check the `this.props.walletType`. Something like this: 
```
this.props.isMobile && (this.props.walletType === 'Mobile' || this.props.walletType === 'Origin Wallet')
```

Reply:
Good point, thanks!

Reply:
Addressed in b6d4a5b.

Reply:
@norbertdragan Sorry if I didn't word it correctly. but I meant to remove the entire code from `componentDidUpdate`. It's complicating things. 

You don't have to hardcode the mobile app links either. `https://originprotocol.com/mobile` will open the corresponding store on mobile.

You could just use `isMobile` and `walletType` props inside the `render` method and render the App download modal.